mm2src/lp_bot/simple_market_maker.rs

```diff
@@ -0,0 +1,266 @@
+    }
+    // todo: cancel all pending orders
+}
+
+async fn update_single_order(_cfg: SimpleCoinMarketMakerCfg, uuid: Uuid, _order: MakerOrder, key_trade_pair: String) {
+    info!("need to update order: {} of {}", uuid, key_trade_pair)
+}
+
+async fn process_bot_logic(ctx: &MmArc) {
+    let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(ctx).unwrap();
+    // note: Copy the cfg here will not be expensive, and this will be thread safe.
+    let cfg = simple_market_maker_bot_ctx.trading_bot_cfg.lock().await.clone();
+
+    let mut memoization_pair_registry: HashSet<String> = HashSet::new();
+    let maker_orders = retrieve_my_maker_orders(ctx).await;
+
+    info!("np_orders: {}", maker_orders.len());
+    for (key, value) in maker_orders.into_iter() {
+        let key_trade_pair = value.base.clone() + "/" + value.rel.clone().as_str();
+        match cfg.get(&key_trade_pair) {
+            Some(coin_cfg) => {
+                update_single_order(coin_cfg.clone(), key, value.clone(), key_trade_pair.clone()).await;
+                memoization_pair_registry.insert(key_trade_pair.clone());
+            },
+            _ => continue,
+        }
+        println!("{}", key);
+    }
+}
+
+pub async fn lp_bot_loop(ctx: MmArc) {
+    info!("lp_bot_loop successfully started");
+    loop {
+        // todo: this log should probably in debug
+        info!("tick lp_bot_loop");
+        if ctx.is_stopping() {
+            // todo: can we cancel all the pending orders when the ctx is stopping or call tear_down ?
+            break;
+        }
+        let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(&ctx).unwrap();
+        let mut states = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+        if states.is_stopping.load(Ordering::Relaxed) {
+            states.is_running = AtomicBool::new(false);
+            states.is_stopping = AtomicBool::new(false);
+            // todo: verify if there is a possible deadlock here if i use states inside tear_down_bot
+            tear_down_bot(ctx).await;
+            break;
+        }
+        drop(states);
+        process_bot_logic(&ctx).await;
+        Timer::sleep(30.0).await;
+    }
+    info!("lp_bot_loop successfully stopped");
+}
+
+pub async fn process_price_request() -> Result<(StatusCode, String, HeaderMap), MmError<PriceServiceRequestError>> {
+    info!("Fetching price from: {}", KMD_PRICE_ENDPOINT);
+    let (status, headers, body) = slurp_url(KMD_PRICE_ENDPOINT).await?;
+    Ok((status, std::str::from_utf8(&body)?.trim().into(), headers))
+}
+
+async fn fetch_price_tickers(ctx: &MmArc) {
+    let (status_code, body, _) = match process_price_request().await {
+        Ok(x) => x,
+        Err(_) => return,
+    };
+    if status_code == StatusCode::OK {
+        // todo: here there is not any chance that the unwrap fail i guess because the status code is OK
+        let model: TickerInfosRegistry = serde_json::from_str(&body).unwrap();
```

Comment:
The server's response format can change or it can return garbage (not valid JSON). Unwrapping on deserialization results can be valid only in tests code.